bin/i18n/i18n_script_utils.rb

```diff
@@ -321,4 +326,48 @@ def self.fix_yml_file(filepath)
 
     File.write(filepath, yml_data)
   end
+
+  # Return true iff the specified file in the specified locale had changes
+  # since the last successful sync-out.
+  #
+  # @param locale [String] the locale code to check. This can be either the
+  #  four-letter code used internally (ie, "es-ES", "es-MX", "it-IT", etc), OR
+  #  the two-letter code used by crowdin, for those languages for which we have
+  #  only a single variation ("it", "de", etc).
+  #
+  # @param file [String] the path to the file to check. Note that this should be
+  #  the relative path of the file as it exists within the locale directory; ie
+  #  "/dashboard/base.yml", "/blockly-mooc/maze.json",
+  #  "/course_content/2018/coursea-2018.json", etc.
+  def self.file_changed?(locale, file)
+    @change_data ||= CROWDIN_PROJECTS.map do |_project_identifier, project_options|
+      # TODO: investigate the condition as a potential cause of sync fails
+      unless File.exist?(project_options[:files_to_sync_out_json])
+        raise <<~ERR
+          File not found #{project_options[:files_to_sync_out_json]}.
+
+          We expect to find a file containing a list of files changed by the most
+          recent sync down; if this file does not exist, it likely means that no
+          sync down has been run on this machine, so there is nothing to sync out
+        ERR
+      end
+      JSON.load_file(project_options[:files_to_sync_out_json])
+    end
+
+    crowdin_code = PegasusLanguages.get_code_by_locale(locale)
+
+    @change_data.any? {|change_data| change_data.dig(locale, file) || change_data.dig(crowdin_code, file)}
+  end
+
+  # Formats strings like 'en-US' to 'en_us'
+  #
+  # @param [String] locale the BCP 47 (IETF language tag) format (e.g., 'en-US')
+  # @return [String] the BCP 47 (IETF language tag) JS format (e.g., 'en_us')
+  def self.to_js_locale(locale)
+    locale.tr('-', '_').downcase
+  end
+
+  def self.create_progress_bar(**args)
+    ProgressBar.create(**args, format: PROGRESS_BAR_FORMAT)
+  end
```

Comment:
Here is some more context to the library used: https://github.com/jfelchner/ruby-progressbar.
Not necessary, but I would add a comment linking to the library and how to use this progress bar, so other teams can use it more easily.